### ISSUE TITLE
Check for Valid C++11 Compiler, Libraries at Pre-compile -time

### ISSUE THREAD

**Issue by jm-janzen** [jm-janzen/termq-engine#50]
See [makefile source here](https://github.com/jm-janzen/termq/blob/master/makefile).  Presently only checks for an environment variable.

I'd like a solution that checks the builder's OS for the required to tools to compile a `C++11`/`ncurses` terminal application.

Ideally, a new makefile would do the following:
Warn that required libraries are missing when a builder tries to run `make`.
(Bonus) Prompt to install missing librar(y|ies).
Check that a valid `C++11` compiler exists on the system, and assign it to `$(CXX)`.

If you discover any other issues with this project's makefile, please note them, and I'll spin up some new issues also with the **hacktoberfest** label.

**Comment by bvbfan** [comment]
Solution is CMAKE, do you want it?

**Comment by jm-janzen** [comment]
Yes, that would be most appreciated.  This is my first c++ application outside of college, so any assistance is welcome.  

Just let me know if/how I can help.

**Comment by shivan1b** [comment]
Hi! I was thinking of adding up the checks and warnings for dependencies in the makefile. Please let me know if I understood it correctly: You want that when user runs make command, warnings about missing libraries/outdated versions should also be shown.
Or if you think that generating Makefile using cmake is a better alternative, you can go with bvbfan's solution. Please inform. Thanks.

**Comment by jm-janzen** [comment]
Hi @shivan1b, firstly thank you for your interest. 

What you describe is indeed what I was looking for.  Unfortunately, I know even less about `cmake` than I do about `make`, so I'm not sure what @bvbfan's solution will look like.  I would be happy to accept PRs from both of you, if only to have two working build options available.

If that sounds alright to you, feel free to start up a new issue, and I'll add the hacktoberfest label to it.

**Comment by shivan1b** [comment]
I think I should submit a PR to this issue as my solution directly addresses it. And maybe start a new issue like "Better way to create Makefile" or something like that. What do you say?

And maybe you will have to assign the issue to me then only it would be counted as a contribution to Hacktoberfest (saw some people commenting like that).

**Comment by jm-janzen** [comment]
@bvbfan per my discussion with shivan1b, I've started a new issue that's more specific to your cmake solution.  Please see it [here](https://github.com/jm-janzen/termq/issues/64).

EDIT:  Did some research, and I don't think issues need to be assigned to users to count.  In fact, the issues don't even need the "hacktoberfest" label (that's just to help pushers find them more easily).  For instance, looking at [bvbfan's contributions](http://hacktoberfestchecker.herokuapp.com/?username=bvbfan) using the hacktoberfest checker web service, all their contributions are without assignment and without the hacktoberfest label.  So it seems you're fine, as long you've signed up for hacktoberfest. :)

**Comment by shivan1b** [comment]
@jm-janzen Thanks for informing. :) Should this issue be closed now since cmake works fine? Or do you still plan on testing the Makefile solution?

**Comment by jm-janzen** [comment]
As long as you don't mind coding it, I would love a dedicated Makefile.
